tests/serialization/test_dag_serialization.py

```diff
@@ -861,6 +892,30 @@ def __init__(self, do_xcom_push=False, **kwargs):
 
         assert serialized_op.do_xcom_push is False
 
+    def test_operator_custom_serialized_fields_present(self):
+        class MyOperator(BaseOperator):
+            __serialized_fields: Optional[frozenset] = None
+
+            def __init__(self, a_dictionary_value: Dict, **kwargs):
+                super().__init__(**kwargs)
+                self.a_dictionary_value = a_dictionary_value
```

Comment:
Why would you want this field in serialized json? if the reason is that it is template_filed, then this can pass right now without any changes to the actual code (only test change) as follows:

```python
    def test_operator_custom_serialized_fields_present(self):
        class MyOperator(BaseOperator):

            template_fields = (
                'a_dictionary_value',
            )

            def __init__(self, a_dictionary_value: Dict, **kwargs):
                super().__init__(**kwargs)
                self.a_dictionary_value = a_dictionary_value

        my_simple_dict = {"a": 1337, "b": 7331}
        op = MyOperator(task_id="dummy", a_dictionary_value=my_simple_dict)
        assert op.a_dictionary_value == my_simple_dict

        blob = SerializedBaseOperator.serialize_operator(op)
        serialized_op = SerializedBaseOperator.deserialize_operator(blob)

        assert serialized_op.a_dictionary_value == my_simple_dict  # type: ignore # pylint: disable=no-member
```

Reply:
This is a valid work around. 
I didn't check that this wouldn't cause issues.

Shall I modify the PR so that we incorporate these tests to make it a bit more visible that this is an option?

Reply:
We already cover a bit of that, by checking `BashOperator`  in `def make_simple_dag` in `tests/serialization/test_dag_serialization.py` adds a template_field (`bash_command`) and we test that it is added in serialized json.

It was added in https://github.com/apache/airflow/pull/7633/files#diff-bb7c036c775e66eeef50932e062fbfc3

I am curious to know if that address you use-case or if not what would be the reason to use DAG serialization in your specific case